File: sdk/src/main/java/com/robotemi/sdk/listeners/OnTelepresenceStatusChangedListener.java
package com.robotemi.sdk.listeners;

import com.robotemi.sdk.telepresence.CallState;

public abstract class OnTelepresenceStatusChangedListener {

    public String sessionId;

    public OnTelepresenceStatusChangedListener(String sessionId) {
        this.sessionId = sessionId;
    }

    /**
     * Called when Telepresence status was changed.
     *
     */
    public abstract void onTelepresenceStatusChanged(CallState callState);
}

File: sdk/src/main/java/com/robotemi/sdk/listeners/OnTelepresenceStatusChangedListener.kt
package com.robotemi.sdk.listeners

import com.robotemi.sdk.telepresence.CallState

/**
 * @param sessionId is no longer available from startTelepresence return value, it is not required any more.
 */
abstract class OnTelepresenceStatusChangedListener(var sessionId: String) {

    /**
     * Called when Telepresence status was changed.
     *
     */
    abstract fun onTelepresenceStatusChanged(callState: CallState)
}
